src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -383,7 +385,7 @@ private void ParseJson(JObject json)
 #pragma warning disable CS0618
                     var spec = JsonPackageSpecReader.GetPackageSpec(specJson);
 #pragma warning restore CS0618
-                    _projects.Add(prop.Name, spec);
+                    _projects.TryAdd(prop.Name, spec);
```

Comment:
The semantics here are probably changing. I'm not sure about `SortedDictionary<TKey, TValue>`, but `Dictionary<TKey, TValue>` throws when you try to call `.Add` on a key that already exists. The new code will silently ignore the new value.

Reply:
Changed the syntax.

Reply:
using `dict[key] = value` does not reproduce the semantics of `Dictionary<T1, T2>()`'s `.Add` method which throws an exception when the key already exists.

I have no idea if any of our code depends on that behaviour, throwing if the key already exists, so maybe it's not a problem. But the changed syntax doesn't maintain semantics.

Reply:
Ok. I added wrapper function `ConcurrentDictAddCheckDuplicateKey` which check for duplicate key before add value and throw exception if it's duplicate key to keep existing behaviour. Please check.